runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -6863,12 +6953,20 @@ static void genInitArrayHeader(
    // for zero sized arrays we unconditionally store 0 in the third dword of the array object header. That is
    // safe because the 3rd dword is either array size of a zero sized array or will contain the first elements
    // of an array:
-   // - Zero sized arrays have the following layout:
+   // - When dual header shape is enabled zero sized arrays have the following layout:
    // - The smallest array possible is a byte array with 1 element which would have a layout:
    //   #bits per section (compressed refs): | 32 bits |  32 bits   | 32 bits | 32 bits |   32 bits   |   32 bits    |
+   //   zero sized arrays:                   |  class  | mustBeZero |  size   | padding |            other           |
+   //   smallest contiguous array:           |  class  |    size    | padding |           other                      |
+   //   This also reflects the minimum object size which is 8 bytes.
```

Comment:
If the minimum object size can change based on policy, keep in mind there are a few places in this file (and in the other backends) that use the configuration macro `J9_GC_MINIMUM_OBJECT_SIZE` (which is presently 16 on x86-64).  If you haven't already, you should find those places that use that macro and confirm that the logic is correct for dual header where the minimum size is 8.

Reply:
Sorry, there was a misunderstanding about minimum object size. `J9_GC_MINIMUM_OBJECT_SIZE` doesn't include `dataAddr` size so we'll never have an indexable object smaller than 16 bytes. I have updated my change to reflect that.